app/Http/Controllers/Api/V1/SearchController.php

```diff
@@ -99,7 +107,19 @@ public function search(SearchRequest $searchRequest)
     public function advance(SearchRequest $searchRequest)
     {
         $data = $searchRequest->validated();
-        $data['indexing'] = [3];
+
+        $organization = Organization::find($data['organization_id']);
+        $this->authorize('advanceSearch', $organization);
+
+        $data['organizationIds'] = [$data['organization_id']];
+
+        if(!auth()->user()->hasPermissionTo('organization:view', $organization)) {
+            $data['organizationVisibilityTypeIds'] = [null, 4];
+            $data['indexing'] = [3];
+        } elseif (isset($data['query'])) {
```

Comment:
formatting issue; need a space after if.
Also, instead of using hardcoded values for visibility types, or role ids, wouldn't the use of constants be better?

Reply:
Added space after "if".
Added constants for "PUBLIC_ORGANIZATION_VISIBILITY_TYPE_ID", "GLOBAL_ORGANIZATION_VISIBILITY_TYPE_ID", "INDEXING_APPROVED" and "INDEXING_OPTIONS".
Replaced hardcoded values with added constants.